checker/tests/resourceleak/TryWithResourcesMultiResources.java

```diff
@@ -0,0 +1,41 @@
+import java.io.IOException;
+import java.net.Socket;
+import org.checkerframework.checker.calledmethods.qual.*;
+import org.checkerframework.checker.mustcall.qual.*;
+
+public class TryWithResourcesMultiResources {
+
+  class OuterResource implements java.io.Closeable {
+    private final @Owning Socket socket;
+
+    public @MustCallAlias OuterResource(@MustCallAlias Socket sock) throws IOException {
+      this.socket = sock;
+    }
+
+    @Override
+    @EnsuresCalledMethods(
+        value = {"this.socket"},
+        methods = {"close"})
+    public void close() throws IOException {
+      this.socket.close();
+    }
+  }
+
+  public void multiResourcesWrong(String address, int port) {
+    // :: error: required.method.not.called
+    try (OuterResource outer = new OuterResource(new Socket(address, port))) {
```

Comment:
Can you add a comment here explaining exactly what's wrong with this version of the test? I don't understand the difference between this version and the version that passes below.

Reply:
Yes we should add a comment.  It's a subtle difference.  try-with-resources can only close resources that have been assigned to a variable.  Here, the call to the `OuterResource` constructor can throw an `IOException`, and if that exception is thrown, the `Socket` will not be closed.  In the other version, the `Socket` is assigned to a separate try-with-resources variable, so it _will_ be closed even if the `OuterResource` constructor throws an exception.

Reply:
The difference is that if `new OuterResource(...)` throws an exception, the reference to the socket goes out of scope without being closed.

Reply:
https://github.com/typetools/checker-framework/pull/4959/commits/7690901a26ac163da9c6877b9a4743c99b10022e